jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py

```diff
@@ -324,7 +325,18 @@ def calculate_moment(self, add_data=True):
         # convert units for moment 0, which is the only currently supported
         # moment for using converted units.
         if n_moment == 0:
-            self.moment = self.moment.to(self.moment_zero_unit)
+            x_unit = u.Unit(self.spectrum_viewer.state.x_display_unit)
+            # only flux<> flux conversions will occur, so we only
+            # need the spectral_density equivalency. should I just
+            # be using the mean wavelength here?
+            eqv = u.spectral_density(np.mean(slab.spectral_axis))
```

Comment:
Would spectral axis ever have NaN (e.g., from masking)?
```suggestion
            eqv = u.spectral_density(np.nanmean(slab.spectral_axis))
```

Reply:
This is an assumption that mean wavelength represents the effective wavelength of a collapsed cube. Does this assumption always hold? If not, the converted value might be misleading. If we cannot promise correctness, maybe better to disallow.

Reply:
that was the assumption i made, but I'm not sure it's correct. is there a better way to calculate an 'effective wavelength' for a cube? (cc @camipacifici)

Reply:
In my INS days, we grabbed the effective wavelength from the header based on the filter used. But that is hard to generalize for arbitrary telescope/instrument. If we must allow unit conversion, we might have to give user the option to manually enter the correct effective wavelength. But I say just disallow it if straight conversion is not possible.

Reply:
I think we should leave the decision to the user and not assume a conversion that might be inaccurate. For narrow subsets, using the median of the wavelength range of the subset should be fine, but for, e.g., collapsing the full wavelength extent of the cube, the two procedures (convert then collapse vs collapse then convert) could return different results. If there is a default and a warning is returned, the user might miss it. Could the offended dataset be grayed out if a conversion of this kind is needed? The user could then decide whether to keep it with the approximate conversion or discard it and create the collapsed image again in the new units.

Reply:
> Could the offended dataset be grayed out if a conversion of this kind is needed?

Grayed out where?  We could (without too much effort) create some sort of infrastructure for warnings/alerts about specific data layers where the legend and data menu get a ‼️ symbol with text in the tooltip, or we could maybe (I'd need to think through the consequences more first) migrate some of these to be auto-updating like spectral extraction and react to a change in units 🤔